src/java.base/share/classes/java/util/BitSet.java

```diff
@@ -1398,4 +1043,405 @@ private int nextSetBit(int fromIndex, int toWordIndex) {
+        }
+
+        /**
+         * @throws IndexOutOfBoundsException {@inheritDoc}
+         */
+        @Override
+        public int previousSetBit(int fromIndex) {
+            return BitSet.previousSetBit(words, words.length, this::length, fromIndex);
+        }
+
+        /**
+         * @throws IndexOutOfBoundsException {@inheritDoc}
+         */
+        @Override
+        public int previousClearBit(int fromIndex) {
+            return BitSet.previousClearBit(words, words.length, fromIndex);
+        }
+
+        @Override
+        public IntStream stream() {
+            // Todo: Improve on this
+            if (isEmpty()) {
+                return IntStream.empty();
+            }
+            return IntStream.iterate(nextSetBit(0), i -> i != -1, this::nextSetBit);
+        }
+
+        @Override
+        public byte[] toByteArray() {
+            return BitSet.toByteArray(words, words.length);
+        }
+
+        @Override
+        public long[] toLongArray() {
+            return words.clone();
+        }
+
+        /**
+         * @throws IndexOutOfBoundsException {@inheritDoc}
+         */
+        @Override
+        public BitSetReadOps get(int fromIndex, int toIndex) {
+            BitSet bs = BitSet.get(words, length(), fromIndex, toIndex);
+            return of(bs);
+        }
+
+        @Override
+        public boolean intersects(BitSet set) {
+            return BitSet.intersects(words, words.length, set.words, set.wordsInUse);
+        }
+
+        @Override
+        public boolean intersects(BitSetReadOps set) {
+            return switch (set) {
+                case BitSet otherBs -> intersects(otherBs);
+                case OfImmutable ro -> BitSet.intersects(words, words.length, ro.words, ro.words.length);
+            };
+        }
+
+        @Override
+        public int size() {
+            return words.length * BITS_PER_WORD;
+        }
+
+        @Override
+        public Object clone() {
+            try {
+                BitSet result = (BitSet) super.clone();
+                // As the array never changes, we can reuse it
+                result.words = words;
+                return result;
+            } catch (CloneNotSupportedException e) {
+                throw new InternalError(e);
+            }
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            return switch (o) {
+                case BitSet bs -> BitSet.equals(words, words.length, bs.words, bs.wordsInUse);
+                case OfImmutable ro -> BitSet.equals(words, words.length, ro.words, ro.length);
+                default -> false;
+            };
+        }
+
+        @Override
+        public int hashCode() {
+            return BitSet.hashCode(words, words.length);
+        }
+
+        @Override
+        public String toString() {
+            return BitSet.toString(this, words.length);
+        }
+
+        private int cardinality0() {
```

Comment:
In this version these two (`cardinality0` and `length0`) just delegate to a simple static method and are only used once each in the constructor - inline them?

Reply:
Good idea.